packages/checkbox/src/react/index.tsx

```diff
@@ -1,42 +1,75 @@
-import { compose, css } from 'glamor'
+import Halo from '@pluralsight/ps-design-system-halo'
+import Theme, { useTheme } from '@pluralsight/ps-design-system-theme'
+import {
+  omit,
+  RefForwardingComponent
+} from '@pluralsight/ps-design-system-util'
+import { compose, css, StyleAttribute } from 'glamor'
 import React from 'react'
-import PropTypes from 'prop-types'
 
-import filterReactProps from '@pluralsight/ps-design-system-filter-react-props'
-import Halo from '@pluralsight/ps-design-system-halo'
-import { useTheme } from '@pluralsight/ps-design-system-theme'
+import stylesheet from '../css'
 
-import stylesheet from '../css/index.js'
+type StyleFn = (
+  themeName: typeof Theme.names,
+  props: CheckboxProps
+) => StyleAttribute
 
 const styles = {
-  checkbox: (themeName, { disabled }) =>
+  checkbox: (themeName, props) =>
     compose(
       css(stylesheet['.psds-checkbox']),
-      disabled && css(stylesheet['.psds-checkbox--disabled'])
+      props.disabled && css(stylesheet['.psds-checkbox--disabled'])
     ),
-  square: (themeName, { checked, indeterminate }) =>
+  square: (themeName, props) =>
     compose(
       css(stylesheet['.psds-checkbox__square']),
       css(stylesheet[`.psds-checkbox__square.psds-theme--${themeName}`]),
-      (checked || indeterminate) &&
+      (props.checked || props.indeterminate) &&
         css(stylesheet['.psds-checkbox__square--active'])
     ),
   icon: () => css(stylesheet['.psds-checkbox__icon']),
   input: () => css(stylesheet['.psds-checkbox__input']),
-  label: themeName =>
+  label: (themeName, _props) =>
     compose(
       css(stylesheet['.psds-checkbox__label']),
       css(stylesheet[`.psds-checkbox__label.psds-theme--${themeName}`])
     )
 }
 
+interface CheckboxProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    Record<string, unknown> {
+  checked?: boolean
+  disabled?: boolean
+  error?: boolean
+  indeterminate?: boolean
+  label: React.ReactNode
+  name?: string
+  onCheck?: (
+    evt: Event,
+    checked: boolean,
+    value: string | number,
```

Comment:
```suggestion
    value: React.ReactText,
```

we've done this with other form elements?